src/layers/domain/core/questionnaire.py

```diff
@@ -27,49 +30,238 @@ class Question(BaseModel):
     name: str
     type: QuestionType
     multiple: bool
+    validation_rules: list[str] = []
+    choices: list[str] = []
 
 
 class Questionnaire(BaseModel):
     """
     A Questionnaire represents a collection of Questions, in a specific order.
     """
 
-    id: UUID
     name: str = Field(regex=ENTITY_NAME_REGEX)
-    _questions: list[Question] = (lambda: [])()
+    version: int
+    questions: list[Question] = []
 
     def __contains__(self, question_name: str) -> bool:
         """
         Returns true if the question specified exists within the questionnaire
         """
-        return question_name in (q.name for q in self._questions)
+
+        return question_name in (q.name for q in self.questions)
 
     def add_question(
         self,
         name: str,
         type: QuestionType = QuestionType.STRING,
         multiple: bool = False,
+        validation_rules: list[str] = [],
+        choices: list[str] = [],
```

Comment:
change to `set` as we don't want duplicates and `set` will offer us some nicer performance